### DISCUSSION TITLE
pydantic model validation

### DISCUSSION THREAD

**Discussion by bergran** [fastapi/fastapi#8150] [Questions] [answered]
Hello! I would like to validate a field on Pydantic, this is not hard. My problem is the next. This field is a `UniqueConstrain` in the database and i have to ask Database with SqlAlchemy if exists, and FastApi has not a system to pass `context` or something like that to share with BaseModel my database session to check it.

How can I validate my unique constrain into Pydantic?

At this time, the solution that i found it, it is to check on the view, but i think it should not do it there, it does?

Thank you!

**Comment by tiangolo** [comment, marked as answer]
You're doing it right. It should be done inside the path operation, or even dependency if you want. That wouldn't belong to part of the validations done by Pydantic.

Those validations are for the data shape and types. The relationships between whatever comes in those shapes and types with your actual database belongs to the next layer, your application code.

I imagine this answers your question, so I'll close this issue. But feel free to add more comments.
